### PR TITLE
Moved AuthorizationTest to groovy spock framework

### PR DESCRIPTION
**Pull request by srujana4**
# What Changed
Moved AuthorizationTest to groovy spock framework
# Why
github issue #37 
Todo:

- [x] Add tests
- [ ] Add docs

### CODE REVIEW

**Comment by srujana4** [comment]
I am having trouble in including these tests to be run as part of the `mvn clean install` @ashpak-shaikh Could you please help

File: src/test/groovy/com/intuit/graphql/authorization/enforcement/AuthorizationTest.groovy [hunk 1]
```diff
@@ -0,0 +1,544 @@
...
+            JsonObject obj = element.getAsJsonObject()
+            if (obj.get("name").getAsString().equals(typeName)) {
+                JsonArray fields = obj.get("fields").getAsJsonArray()
+                return fields.stream().map { it.getAsJsonObject().get("name").getAsString() }.collect(Collectors.toList())
+            }
+        }
+        return []
+    }
+
+}
```
Comment (ashpak-shaikh) [inline src/test/groovy/com/intuit/graphql/authorization/enforcement/AuthorizationTest.groovy:544]: Some comments on groovy style tests 

1. most things should be done inline with multiline strings, like all the files can be kept inline within the test.
2. Groovy tests are less verbose, example maps can be initialized directly
3. assertThat is not needed

Check tests here for example
https://github.com/graph-quilt/graphql-orchestrator-java/tree/master/src/test/groovy/com/intuit/graphql/orchestrator/integration

**Comment by ashpak-shaikh** [comment]
This is good, you just need the gmaven plugin and also be sure to include the Specification file in the surefire plugin. 

https://stackoverflow.com/questions/43824908/maven-plugin-for-groovy-tests

**Comment by ashpak-shaikh** [comment]
```
 Tests run: 20, Failures: 0, Errors: 20, Skipped: 0, Time elapsed: 0.806 s <<< FAILURE! - in com.intuit.graphql.authorization.enforcement.AuthorizationSpec
Error:  com.intuit.graphql.authorization.enforcement.AuthorizationSpec.test authorization with some redactions with list  Time elapsed: 0.058 s  <<< ERROR!
java.lang.IllegalAccessException: class org.spockframework.util.ReflectionUtil cannot access a member of class com.intuit.graphql.authorization.enforcement.AuthorizationSpec with modifiers "private"

Error:  com.intuit.graphql.authorization.enforcement.AuthorizationSpec.test authz with no client configuration  Time elapsed: 0.003 s  <<< ERROR!
java.lang.IllegalAccessException: class org.spockframework.util.ReflectionUtil cannot access a member of class com.intuit.graphql.authorization.enforcement.AuthorizationSpec with modifiers "private"

Error:  com.intuit.graphql.authorization.enforcement.AuthorizationSpec.test authz Introspection With Some Redactions  Time elapsed: 0.001 s  <<< ERROR!
java.lang.IllegalAccessException: class org.spockframework.util.ReflectionUtil cannot access a member of class com.intuit.graphql.authorization.enforcement.AuthorizationSpec with modifiers "private"

Error:  com.intuit.graphql.authorization.enforcement.AuthorizationSpec.test authz happy case  Time elapsed: 0.004 s  <<< ERROR!
java.lang.IllegalAccessException: class org.spockframework.util.ReflectionUtil cannot access a member of class com.intuit.graphql.authorization.enforcement.AuthorizationSpec with modifiers "private"

Error:  com.intuit.graphql.authorization.enforcement.AuthorizationSpec.test authz happy case all fields  Time elapsed: 0.001 s  <<< ERROR!
java.lang.IllegalAccessException: class org.spockframework.util.ReflectionUtil cannot access a member of class com.intuit.graphql.authorization.enforcement.AuthorizationSpec with modifiers "private"

Error:  com.intuit.graphql.authorization.enforcement.AuthorizationSpec.test authz happy case all fields with fragments  Time elapsed: 0.002 s  <<< ERROR!
java.lang.IllegalAccessException: class org.spockframework.util.ReflectionUtil cannot access a member of class com.intuit.graphql.authorization.enforcement.AuthorizationSpec with modifiers "private"

Error:  com.intuit.graphql.authorization.enforcement.AuthorizationSpec.test no Authz  Time elapsed: 0.002 s  <<< ERROR!
java.lang.IllegalAccessException: class org.spockframework.util.ReflectionUtil cannot access a member of class com.intuit.graphql.authorization.enforcement.AuthorizationSpec with modifiers "private"

Error:  com.intuit.graphql.authorization.enforcement.AuthorizationSpec.test authz with invalid scope  Time elapsed: 0.001 s  <<< ERROR!
java.lang.IllegalAccessException: class org.spockframework.util.ReflectionUtil cannot access a member of class com.intuit.graphql.authorization.enforcement.AuthorizationSpec with modifiers "private"

Error:  com.intuit.graphql.authorization.enforcement.AuthorizationSpec.test authz multi scopes  Time elapsed: 0.001 s  <<< ERROR!
java.lang.IllegalAccessException: class org.spockframework.util.ReflectionUtil cannot access a member of class com.intuit.graphql.authorization.enforcement.AuthorizationSpec with modifiers "private"

Error:  com.intuit.graphql.authorization.enforcement.AuthorizationSpec.test authz with invalid field  Time elapsed: 0.002 s  <<< ERROR!
java.lang.IllegalAccessException: class org.spockframework.util.ReflectionUtil cannot access a member of class com.intuit.graphql.authorization.enforcement.AuthorizationSpec with modifiers "private"

Error:  com.intuit.graphql.authorization.enforcement.AuthorizationSpec.test authz with mutation  Time elapsed: 0.001 s  <<< ERROR!
java.lang.IllegalAccessException: class org.spockframework.util.ReflectionUtil cannot access a member of class com.intuit.graphql.authorization.enforcement.AuthorizationSpec with modifiers "private"

Error:  com.intuit.graphql.authorization.enforcement.AuthorizationSpec.test authz with mutation multi scopes  Time elapsed: 0.001 s  <<< ERROR!
java.lang.IllegalAccessException: class org.spockframework.util.ReflectionUtil cannot access a member of class com.intuit.graphql.authorization.enforcement.AuthorizationSpec with modifiers "private"

Error:  com.intuit.graphql.authorization.enforcement.AuthorizationSpec.test authz with mutation multi scopes2  Time elapsed: 0.002 s  <<< ERROR!
java.lang.IllegalAccessException: class org.spockframework.util.ReflectionUtil cannot access a member of class com.intuit.graphql.authorization.enforcement.AuthorizationSpec with modifiers "private"

Error:  com.intuit.graphql.authorization.enforcement.AuthorizationSpec.test authz with mutation no access  Time elapsed: 0 s  <<< ERROR!
java.lang.IllegalAccessException: class org.spockframework.util.ReflectionUtil cannot access a member of class com.intuit.graphql.authorization.enforcement.AuthorizationSpec with modifiers "private"

Error:  com.intuit.graphql.authorization.enforcement.AuthorizationSpec.test authz with mutation no scope  Time elapsed: 0.003 s  <<< ERROR!
java.lang.IllegalAccessException: class org.spockframework.util.ReflectionUtil cannot access a member of class com.intuit.graphql.authorization.enforcement.AuthorizationSpec with modifiers "private"

Error:  com.intuit.graphql.authorization.enforcement.AuthorizationSpec.test authz with mutation and fragments  Time elapsed: 0.003 s  <<< ERROR!
java.lang.IllegalAccessException: class org.spockframework.util.ReflectionUtil cannot access a member of class com.intuit.graphql.authorization.enforcement.AuthorizationSpec with modifiers "private"

Error:  com.intuit.graphql.authorization.enforcement.AuthorizationSpec.test authz with mutation non oauth2  Time elapsed: 0.001 s  <<< ERROR!
java.lang.IllegalAccessException: class org.spockframework.util.ReflectionUtil cannot access a member of class com.intuit.graphql.authorization.enforcement.AuthorizationSpec with modifiers "private"

Error:  com.intuit.graphql.authorization.enforcement.AuthorizationSpec.test introspection with test client2  Time elapsed: 0.001 s  <<< ERROR!
java.lang.IllegalAccessException: class org.spockframework.util.ReflectionUtil cannot access a member of class com.intuit.graphql.authorization.enforcement.AuthorizationSpec with modifiers "private"

Error:  com.intuit.graphql.authorization.enforcement.AuthorizationSpec.test introspection without scope  Time elapsed: 0.001 s  <<< ERROR!
java.lang.IllegalAccessException: class org.spockframework.util.ReflectionUtil cannot access a member of class com.intuit.graphql.authorization.enforcement.AuthorizationSpec with modifiers "private"

Error:  com.intuit.graphql.authorization.enforcement.AuthorizationSpec.test introspection with multi scopes  Time elapsed: 0.002 s  <<< ERROR!
java.lang.IllegalAccessException: class org.spockframework.util.ReflectionUtil cannot access a member of class com.intuit.graphql.authorization.enforcement.AuthorizationSpec with modifiers "private"
```

**Review by ashpak-shaikh — APPROVED** [review]

**Pull request merged**